GANDLF/utils/gandlf_logger.py

```diff
@@ -0,0 +1,61 @@
+import logging
+import yaml
+from pathlib import Path
+from importlib import resources
+import colorlog
+
+
+def gandlf_logger_setup(log_dir=None, config_path="logging_config.yaml"):
+    """
+    It sets up the logger. Reads from logging_config.
+    If log_dir is None, the logs are flashed to console.
+    Args:
+        log_dir (str): dir path for saving the logs
+        config_path (str): file path for the configuration
```

Comment:
I would suggest to pass a full path to `log_file` instead of a `log_dir`, because if we create default file names, there is a good chance for us to overwrite/append different logs, which can make things tougher for the end user.

```suggestion
def gandlf_logger_setup(log_file=None, config_path="logging_config.yaml"):
    """
    It sets up the logger. Reads from logging_config.
    Args:
        log_file (str): dir path for saving the logs, defaults to `None`, at which time logs are flushed to console.
        config_path (str): file path for the configuration
```